spring-integration-r2dbc/src/main/java/org/springframework/integration/r2dbc/outbound/R2dbcMessageHandler.java

```diff
@@ -0,0 +1,243 @@
+
+	@Nullable
+	private Expression criteriaExpression;
+
+	private Expression queryTypeExpression = new ValueExpression<>(Type.INSERT);
+
+
+	/**
+	 * Construct this instance using a fully created and initialized instance of provided
+	 * {@link R2dbcEntityOperations}
+	 *
+	 * @param r2dbcEntityOperations The R2dbcEntityOperations implementation.
+	 */
+
+	public R2dbcMessageHandler(R2dbcEntityOperations r2dbcEntityOperations) {
+		Assert.notNull(r2dbcEntityOperations, "'r2dbcEntityOperations' must not be null");
+		this.r2dbcEntityOperations = r2dbcEntityOperations;
+	}
+
+
+	public void setQueryType(R2dbcMessageHandler.Type type) {
+		setQueryTypeExpression(new ValueExpression<>(type));
+	}
+
+	public void setQueryTypeExpression(Expression queryTypeExpression) {
+		Assert.notNull(queryTypeExpression, "'queryTypeExpression' must not be null");
+		this.queryTypeExpression = queryTypeExpression;
+	}
+
+	public void setTableName(String tableName) {
+		setTableNameExpression(new LiteralExpression(tableName));
+	}
+
+	public void setTableNameExpression(Expression tableNameExpression) {
+		this.tableNameExpression = tableNameExpression;
+	}
+
+	public void setValuesExpression(Expression valuesExpression) {
+		this.valuesExpression = valuesExpression;
+	}
+
+	public void setCriteriaExpression(Expression criteriaExpression) {
+		this.criteriaExpression = criteriaExpression;
+	}
+
+
+	@Override
+	public String getComponentType() {
+		return "r2dbc:reactive-outbound-channel-adapter";
+	}
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+
+		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+		TypeLocator typeLocator = this.evaluationContext.getTypeLocator();
+		if (typeLocator instanceof StandardTypeLocator) {
+			//Register R2dbc criteria API package so FQCN can be avoided in query-expression.
+			((StandardTypeLocator) typeLocator).registerImport("org.springframework.data.relational.core.query");
+		}
+		this.initialized = true;
+
+	}
+
+	@Override
+	protected Mono<Void> handleMessageInternal(Message<?> message) {
+		Assert.isTrue(this.initialized, "This class is not yet initialized. Invoke its afterPropertiesSet() method");
+		return Mono.fromSupplier(() -> this.queryTypeExpression.getValue(this.evaluationContext, message, Type.class))
+				.flatMap(mode -> {
+					switch (mode) {
+						case INSERT:
+							return handleInsert(message);
+						case UPDATE:
+							return handleUpdate(message);
+						case DELETE:
+							return handleDelete(message);
+						default:
+							return Mono.error(new IllegalArgumentException());
+					}
+
+				}).then();
+	}
+
+
+	private Mono<Void> handleDelete(Message<?> message) {
+		if (this.tableNameExpression != null) {
+			String tableName = evaluateTableNameExpression(message);
+			Criteria criteria = evaluateCriteriaExpression(message);
+			DatabaseClient.DeleteMatchingSpec deleteSpec = this.r2dbcEntityOperations.getDatabaseClient().delete().from(tableName);
```

Comment:
The line length should not be longer than 120 symbols: much easier to review and read code flow.
Also I prefer 
the
.method
.chain
.style